src/java.base/share/classes/jdk/internal/module/ModuleBootstrap.java

```diff
@@ -469,15 +468,27 @@ private static ModuleLayer boot2() {
                 limitedFinder = new SafeModuleFinder(finder);
         }
 
+        // if -Xshare:dump and mainModule are specified, check if the mainModule
+        // is a builtin JDK module. If so, set canArchive to true so that the
```

Comment:
There is no notion of "built-in JDK module" so I think change the comment to say that it checks if the main module is in the run-time image and not on the upgrade module path.

URI schemes are compared without regard to case so it should be equalsIgnoreCase("jrt").

Reply:
Fixed both.